src/trace.rs

```diff
@@ -688,3 +716,34 @@ impl MakeRequestId for UuidMakeRequestId {
         Some(RequestId::new(request_id))
     }
 }
+
+struct HoneycombTraceOnSentryScope {
+    team: String,
+    dataset: String,
+    env: String,
+}
+
+impl HoneycombTraceOnSentryScope {
+    pub fn new(team: String, dataset: String, env: String) -> HoneycombTraceOnSentryScope {
+        HoneycombTraceOnSentryScope { team, dataset, env }
+    }
+}
+
+impl<S: Subscriber> TracingLayer<S> for HoneycombTraceOnSentryScope {
+    fn on_enter(&self, _id: &Id, _ctx: tracing_subscriber::layer::Context<'_, S>) {
+        let context = Span::current().context();
+        let span = context.span();
+        let span_context = span.span_context();
+
+        let trace_id = span_context.trace_id().to_string();
+
+        if trace_id != "00000000000000000000000000000000" {
```

Comment:
I think we need to check if the environment is `staging` or `production` before adding the tag because these are the only envs with Honeycomb support.

Reply:
Done: https://github.com/cloudwalk/balthazar/pull/21/commits/9496b2e74b521c70b3c54e9dad1fbe4444162f05